packages/ui-extensions-sdk/src/types.ts

```diff
@@ -341,3 +341,18 @@ export interface DeprecatedEventUsage {
 }
 
 export type DeprecatedUsage = DeprecatedEventUsage; // we can union type here later
+
+export interface LoadedResourceMetadata {
+    startTimeTs: number; // timestamp (in ms) of the resource started being fetched
+    secureConnectionStartTs: number; // timestamp (in ms) of the SSL handshake
+    url?: string; // the resources URL (set to null in case of network calls for privacy reason). This value doesn't change even if the request is redirected.
+    urlHostname: string; // the resource URL hostname
+    initiatorType: string; // the type of resource that initiated the performance event (xmlhttprequest, css, img)
+    nextHopProtocol: string; // the network protocol used to fetch the resource, as identified by the ALPN Protocol ID (RFC7301).
+    duration: number; // overall time required to fetch the resource
+    decodedBodySize: number; // size of the body after removing any applied content-codings
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏If these comments are turned into JSDoc format, they will show up in editors (like VS Code). Not blocking, so up to you whether you want to change it or not.